frontend/src/pages/projects/screens/spawner/environmentVariables/useNotebookEnvVariables.tsx

```diff
@@ -61,17 +75,26 @@ export const fetchNotebookEnvVariables = (notebook: NotebookKind): Promise<EnvVa
 
 export const useNotebookEnvVariables = (
   notebook?: NotebookKind,
-): [envVariables: EnvVariable[], setEnvVariables: (envVars: EnvVariable[]) => void] => {
+): [
+  envVariables: EnvVariable[],
+  setEnvVariables: (envVars: EnvVariable[]) => void,
+  envVariableLoaded: boolean,
+] => {
   const [envVariables, setEnvVariables] = React.useState<EnvVariable[]>([]);
 
-  React.useEffect(() => {
-    if (notebook) {
-      fetchNotebookEnvVariables(notebook)
-        .then((envVars) => setEnvVariables(envVars))
-        /* eslint-disable-next-line no-console */
-        .catch((e) => console.error('Reading environment variables failed: ', e));
+  const callback = React.useCallback(() => {
+    if (!notebook) {
+      return Promise.reject(new NotReadyError('No notebook'));
     }
+
+    return fetchNotebookEnvVariables(notebook);
   }, [notebook]);
 
-  return [envVariables, setEnvVariables];
+  const [fetchedVariable, envVariableloaded] = useFetchState(callback, []);
+
+  React.useEffect(() => {
+    setEnvVariables(fetchedVariable);
+  }, [fetchedVariable]);
```

Comment:
There is a very small issue with this method that could lead to a flickering with the alert. Imagine the following situation: You have some values in `notebook.spec.template.spec.containers[0].envFrom` and the secrets/config maps are not deleted. Then you will call `useNotebookEnvVariables` hook. The `envVariableLoaded` will become `true` as soon as the fetch finishes, while there is a very short time before it re-renders and `setEnvVariables` here. That will cause a flickering to show the alert very quickly and disappear. Although it's very hard to notice that and it normally won't have any issue, it's still better to get fixed if there is a way to do so. Maybe use a second loaded state and `useRef` to set it? 🤔